components/Settings/FavoritesTab.js

```diff
@@ -1,9 +1,19 @@
+import { useState } from 'react'
 import ClubTableRow from '../ClubTableRow'
 
-export default (props) => {
-  const { keyword, updateFavorites, favorites, clubs } = props
-
+export default ({ keyword, clubs, favorites, updateFavorites }) => {
+  const [table, setTable] = useState(() => {
+    const ret = {}
+    favorites.forEach(favorite => {
+      ret[favorite] = true
+    })
+    return ret
+  })
   const findClub = clubs ? code => clubs.find(club => club.code === code) : () => {}
```

Comment:
Feel like there's a better way to do this—even just splitting into lines and adding a comment that "if there is no code, there is no club to be found" would be a lot more readable than this ternary operator

Reply:
rewritten in the commit "rewrite findClub to be clearer"